internal/pkgmanager/download.go

```diff
@@ -5,24 +5,40 @@ import (
 	"io"
 	"net/http"
 	"os"
-	"path"
 	"path/filepath"
+	"strings"
+
+	"github.com/ossf/package-analysis/internal/utils"
 )
 
 /*
-downloadToDirectory downloads a file from the given URL to the given directory. The name of the
-local file is obtained from the last element of the URL path when split on the '/' character.
+downloadToDirectory downloads a file from the given URL to the given directory.
 On successful download, the full path to the downloaded file is returned.
+
+fileName argument allows passing a custom filename for local file.
+The default filename will be obtained from the last element
+of the URL path when split on the '/' character.
```

Comment:
This comment should be updated. The comment should say that `filename` argument is now required with no default value, and the hash of the file is appended to the given filename. If an error occurs during hashing, then the original filename is used.